internal/common/utils/sets.go

```diff
@@ -51,7 +51,7 @@ func (s StringSet) Size() int {
 
 // ToSlice returns all the elements of StringSet as a slice of strings
 func (s StringSet) ToSlice() []string {
-	result := make([]string, s.Size())
+	result := make([]string, 0, s.Size())
```

Comment:
This was a bug, which manifested when I tried to use this method to find unique AZs.

It would create a slice which looked like this if the set was <entry1, entry2> : ["", "", "entry1", "entry2"] as the second parameter to make is the number of initial elements, the third optional parameter is the capacity.